pkg/csi/service/vanilla/controller.go

```diff
@@ -760,6 +757,22 @@ func (c *controller) ControllerExpandVolume(ctx context.Context, req *csi.Contro
 	log := logger.GetLogger(ctx)
 	log.Infof("ControllerExpandVolume: called with args %+v", *req)
 
+	var volumePath string
+	if strings.Contains(req.VolumeId, ".vmdk") {
+		volumePath = req.VolumeId
+	}
+	if volumePath != "" {
+		if containerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIMigration) {
+			// Migration feature switch is enabled
+			msg := fmt.Sprintf("Cannot expand migrated vSphere volume. :%q", volumePath)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Unimplemented, msg)
+		}
+		// Migration feature switch is disabled and volumePath is observed
+		msg := fmt.Sprintf("volume-migration feature switch is disabled. Cannot use volume with vmdk path :%q", volumePath)
+		log.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
```

Comment:
Why not throw Unimplemented error here so that external resizer can stop continuously calling ControllerExpandVolume?

Reply:
Here I was checking if the volume-migration feature is enabled not, but now regardless of this check, feature is not implemented when vmdk path is presented, so we can simply return Unimplemented regardless of feature is enabled or not.

I will simplify this with

```
	if strings.Contains(req.VolumeId, ".vmdk") {
		msg := fmt.Sprintf("Cannot expand migrated vSphere volume. :%q", req.VolumeId)
		log.Error(msg)
		return nil, status.Errorf(codes.Unimplemented, msg)
	}
```